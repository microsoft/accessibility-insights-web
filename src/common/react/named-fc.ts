// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

export type ReactFCWithDisplayName<P = {}> = React.FC<React.PropsWithChildren<P>> & {
    displayName: string;
};

export function NamedFC<P = {}>(
    displayName: string,
    component: React.FC<React.PropsWithChildren<P>>,
): ReactFCWithDisplayName<P> {
    component.displayName = displayName;

    return component as ReactFCWithDisplayName<P>;
}
