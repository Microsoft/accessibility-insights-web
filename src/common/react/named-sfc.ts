// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

export type ReactSFCWithDisplayName<P = {}> = React.SFC<P> & { displayName: string };

export function NamedSFC<P = {}>(displayName: string, component: React.SFC<P>): ReactSFCWithDisplayName<P> {
    component.displayName = displayName;

    return component as ReactSFCWithDisplayName<P>;
}
