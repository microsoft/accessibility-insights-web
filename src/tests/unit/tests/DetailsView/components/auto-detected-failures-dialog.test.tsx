// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import {
    TabStopRequirementState,
    VisualizationScanResultData,
} from 'common/types/store-data/visualization-scan-result-data';
import {
    AutoDetectedFailuresDialog,
    AutoDetectedFailuresDialogProps,
} from 'DetailsView/components/auto-detected-failures-dialog';
import { shallow } from 'enzyme';
import { PrimaryButton } from 'office-ui-fabric-react';
import * as React from 'react';

describe('AutoDetectedFailuresDialog', () => {
    let props: AutoDetectedFailuresDialogProps;
    let visualizationScanResultData: VisualizationScanResultData;
    const prevState = { autoDetectedFailuresDialogEnabled: false };
    const prevProps = {
        visualizationScanResultData: { tabStops: { tabbingCompleted: false } },
    };

    beforeEach(() => {
        const requirements: TabStopRequirementState = {
            'input-focus': {
                instances: [
                    { id: 'test-id-1', description: 'test desc 1' },
                    { id: 'test-id-2', description: 'test desc 2' },
                ],
                status: 'fail',
                isExpanded: false,
            },
        };
        visualizationScanResultData = {
            tabStops: { tabbingCompleted: true, requirements },
        } as VisualizationScanResultData;
        props = {
            visualizationScanResultData,
        } as AutoDetectedFailuresDialogProps;
    });

    it('renders when dialog is not enabled', () => {
        const wrapper = shallow(<AutoDetectedFailuresDialog {...props} />);

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    describe('on dialog enabled', () => {
        let wrapper;
        beforeEach(() => {
            wrapper = shallow(<AutoDetectedFailuresDialog {...props} />);
            wrapper.instance().componentDidUpdate(prevProps, prevState);
        });

        it('renders when dialog is enabled', () => {
            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('is dismissed when "got it" button is clicked', () => {
            wrapper.find(PrimaryButton).simulate('click');

            expect(wrapper.getElement()).toMatchSnapshot();
        });
    });

    describe('on componentDidUpdate', () => {
        it('displays dialog on tabbing completed', () => {
            const wrapper = shallow(<AutoDetectedFailuresDialog {...props} />);
            expect(wrapper.state('dialogEnabled')).toBe(false);
            wrapper.instance().componentDidUpdate(prevProps, prevState);
            expect(wrapper.state('dialogEnabled')).toBe(true);
        });

        it('does not display dialog with no results', () => {
            visualizationScanResultData.tabStops.requirements = null;

            const wrapper = shallow(<AutoDetectedFailuresDialog {...props} />);
            expect(wrapper.state('dialogEnabled')).toBe(false);
            wrapper.instance().componentDidUpdate(prevProps, prevState);
            expect(wrapper.state('dialogEnabled')).toBe(false);
        });
    });
});
