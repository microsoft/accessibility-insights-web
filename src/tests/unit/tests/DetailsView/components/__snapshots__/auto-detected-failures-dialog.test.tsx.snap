// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoDetectedFailuresDialog on dialog enabled is dismissed when "got it" button is clicked 1`] = `null`;

exports[`AutoDetectedFailuresDialog on dialog enabled renders when dialog is enabled 1`] = `
<BlockingDialog
  dialogContentProps={
    Object {
      "title": "Possible failures auto-detected",
      "type": 0,
    }
  }
  hidden={false}
  modalProps={
    Object {
      "containerClassName": "insightsDialogMainOverride",
    }
  }
>
  <div
    className="dialogBody"
  >
    <ul>
      <li>
        Review each finding to be sure it's valid
      </li>
      <li>
        Add any more failures you discovered manually
      </li>
    </ul>
  </div>
  <StyledDialogFooterBase>
    <CustomizedPrimaryButton
      onClick={[Function]}
      text="Got it"
    />
  </StyledDialogFooterBase>
</BlockingDialog>
`;

exports[`AutoDetectedFailuresDialog renders when dialog is not enabled 1`] = `null`;
